lib/util/cache/package/decorator.ts

```diff
@@ -66,21 +68,59 @@ export function cache<T>({
       return callback();
     }
 
-    const cachedResult = await packageCache.get<unknown>(
+    finalKey = `cache-decorator:${finalKey}`;
+    const cacheRecord = await packageCache.get<DecoratorCachedRecord<T>>(
       finalNamespace,
       finalKey
     );
 
-    if (cachedResult !== undefined) {
-      return cachedResult;
+    const now = DateTime.local();
+    const ttlHardMinutes = ttlMinutes * 10;
+
+    let oldData: T | undefined;
+    if (cacheRecord) {
+      const softDeadline = DateTime.fromISO(cacheRecord.cachedAt).plus({
```

Comment:
we should use an experimental environment values for this, like we do for npm

https://docs.renovatebot.com/self-hosted-experimental/#renovate_cache_npm_minutes